src/scripts/import-needs-assessment-data/add-surveys.ts

```diff
@@ -0,0 +1,288 @@
+    uniqueSurveys.map((survey) => {
+      return new Promise<SurveyUploadWorkflow>((resolve, _reject) => {
+        resolve({
+          data: {
+            year: survey.slice(0, -3),
+            quarter: survey.slice(5, 7),
+          },
+          orig: survey,
+          status: UploadWorkflowStatus.PROCESSING,
+          logs: [],
+        });
+      })
+        .then(parseSurvey)
+        .then(getSurvey)
+        .then(uploadSurvey);
+    }),
+  );
+
+  // { "SUCCESS": [], "ALREADY_EXITS": [], ...}
+  const resultsMap: SurveyUploadWorkflowResults = Object.keys(
+    UploadWorkflowStatus,
+  ).reduce((resultsMap, key) => {
+    resultsMap[key] = [];
+    return resultsMap;
+  }, {} as SurveyUploadWorkflowResults);
+
+  results
+    .map((result) => {
+      if (isFulfilled(result)) {
+        return result.value;
+      } else {
+        return result.reason;
+      }
+    })
+    .reduce((resultsMap, workflowResult) => {
+      if (workflowResult.status) {
+        resultsMap[workflowResult.status].push(workflowResult);
+      } else {
+        resultsMap[UploadWorkflowStatus.OTHER].push(workflowResult);
+      }
+      return resultsMap;
+    }, resultsMap);
+
+  console.log("Add NeedsAssessment.Survey results:");
+  Object.keys(resultsMap).forEach((key) => {
+    console.log(`     ${key}: ${resultsMap[key].length}`);
+
+    // NOTE: uncomment & set the status key to debug different types of results
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validSurveys: Survey[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((surveys: Survey[], workflow: SurveyUploadWorkflow) => {
+    return [...surveys, workflow.data];
+  }, [] as Survey[]);
+
+  return validSurveys;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/*  Consolidate Surveys
+ * ------------------------------------------------------- */
+function consolidateSurveys(data: NeedAssessment[]): string[] {
+  const uniqueSurveys = new Set<string>();
+
+  data.forEach((item) => {
+    const surveyString = `${item.survey.year}-${item.survey.quarter}`;
+    uniqueSurveys.add(surveyString);
+  });
+
+  return Array.from(uniqueSurveys);
+}
+
+/*  Parse Surveys
+ * ------------------------------------------------------- */
+function parseSurvey({
+  data,
+  orig,
+  status,
+  logs,
+}: SurveyUploadWorkflow): SurveyUploadWorkflow {
+  logs = [...logs, `Log: parsing survey "${orig}"`];
+
+  if (typeof orig === "string") {
```

Comment:
I believe I was running into issues again with types here for original testing of functions on their own, and kept it in to cover any event with modified code that resulted in objects instead of strings. I've corrected this to only deal with strings moving forward.